### DISCUSSION TITLE
Difference between next.js api responseLimit vs vercel serverless function limits

### DISCUSSION THREAD

**Discussion by jeevanpillay0000** [vercel/community#1144] [Help]
### Summary

Background

NextJS documentation on [API Routes Response Size Limited to 4MB](https://nextjs.org/docs/messages/api-routes-response-size-limit)
```
If you are not using Next.js in a serverless environment, and understand the performance implications of not using a CDN or dedicated media host, you can set this limit to false inside your API Route.
```

First question, what are situations where we don't use "Next.js in a serverless environment"? 

From my current understanding the API folder has to be used with Vercel to enable serverless usage of the functions. Hence, could someone give any example of where this `/api` folder can be used outside the context of a serverless environment?

----
Vercel on [Serverless Function Payload Size Limit](https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit)

This states that vercel's max payload size is 4.5mb.

Now, if my API config on next.js api is something like
```js
export const config = {
  api: {
    responseLimit: '10mb',
  },
}
```

Would this not work in vercel's serverless functions since it does not adhere to vercel's payload size.

**Comment by swarnava** [comment]
Hi,

For Next.js route, we took the approach of limiting it slightly below the Serverless limit. Howver, your payload shouldn't really get exceeded than 4mb. It generally indicates something wrong with the codebase.

**Comment by rnbrady** [comment]
> First question, what are situations where we don't use "Next.js in a serverless environment"?

You might have your own server (e.g. Ubuntu server on EC2 or in a Kubernetes cluster) where you run `npm build && npm start`.  That would be serverful Next.js and the API routes will work fine that way. They don't require a severless environment.
